authmodules/TYPO3/privacyidea/Classes/PrivacyideaService.php

```diff
@@ -110,23 +113,30 @@ public function init() {
 	 * @return int authentication statuscode, one of 0, 100 and 200
 	 */
 	public function authUser(array $user) {
-		// 0 means authentication failure
-		$ret = 0;
-		$username = $this->login['uname'];
-		$password = $this->login['uident_text'];
-		$this->logger->info("try to authenticate user [$username]");
+		if ($this->extConf['privacyIDEABackend'] != 'adminOnly'
+			|| (bool)$user['admin']
+			|| $this->authInfo['loginType'] == 'FE'
+		) {
```

Comment:
OK, you want 

  1. the frontend user to authenticate with 2 factors.
  2. All backend users with 2 factors, if "always" is set.
  3. Only backend admins with 2factor, it "adminonly" is set.

What is the sense in have non-admin users NOT authenticate with 2 factors but with their normal password?
I would not distinguish this, because, if you want to, you can distinguish this in privacyIDEA.
If you do not enroll tokens for the non-admin-BE-users, than these users could authenticate with their password.

In my opinion it is more straight forward to say:

  1. should BE users use privacyIDEA
  2. should FE users use privacyIDEA

and leave the more complicated logic to privacyidea.